x/auth/keeper/deterministic_test.go

```diff
@@ -0,0 +1,112 @@
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	moduletestutil "github.com/cosmos/cosmos-sdk/types/module/testutil"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/stretchr/testify/suite"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	"pgregory.net/rapid"
+)
+
+type DeterministicTestSuite struct {
+	suite.Suite
+
+	ctx sdk.Context
+
+	queryClient   types.QueryClient
+	accountKeeper keeper.AccountKeeper
+	encCfg        moduletestutil.TestEncodingConfig
+}
+
+func TestDeterministicTestSuite(t *testing.T) {
+	suite.Run(t, new(DeterministicTestSuite))
+}
+
+func (suite *DeterministicTestSuite) SetupTest() {
+	suite.encCfg = moduletestutil.MakeTestEncodingConfig(auth.AppModuleBasic{})
+
+	key := sdk.NewKVStoreKey(types.StoreKey)
+	testCtx := testutil.DefaultContextWithDB(suite.T(), key, sdk.NewTransientStoreKey("transient_test"))
+	suite.ctx = testCtx.Ctx.WithBlockHeader(tmproto.Header{})
+
+	maccPerms := map[string][]string{
+		"fee_collector":          nil,
+		"mint":                   {"minter"},
+		"bonded_tokens_pool":     {"burner", "staking"},
+		"not_bonded_tokens_pool": {"burner", "staking"},
+		multiPerm:                {"burner", "minter", "staking"},
+		randomPerm:               {"random"},
+	}
+
+	suite.accountKeeper = keeper.NewAccountKeeper(
+		suite.encCfg.Codec,
+		key,
+		types.ProtoBaseAccount,
+		maccPerms,
+		"cosmos",
+		types.NewModuleAddress("gov").String(),
+	)
+
+	queryHelper := baseapp.NewQueryServerTestHelper(suite.ctx, suite.encCfg.InterfaceRegistry)
+	types.RegisterQueryServer(queryHelper, suite.accountKeeper)
+	suite.queryClient = types.NewQueryClient(queryHelper)
+}
+
+func (suite *DeterministicTestSuite) runIterations(addr sdk.AccAddress, prevRes types.AccountI, checkPrevRes bool) {
+	for i := 0; i < 1000; i++ {
+		acc, err := suite.queryClient.Account(suite.ctx, &types.QueryAccountRequest{Address: addr.String()})
+		suite.Require().NoError(err)
+		suite.Require().NotNil(acc)
+		var account types.AccountI
+
+		err = suite.encCfg.InterfaceRegistry.UnpackAny(acc.Account, &account)
+		suite.Require().NoError(err)
+		suite.Require().Equal(account.GetAddress(), addr)
+
+		// check with previous response too.
+		// suite.Require().Equal(account.GetAddress(), prevRes.GetAddress())
+		// suite.Require().Equal(account.GetPubKey(), prevRes.GetPubKey())
+		// suite.Require().Equal(account.GetSequence(), prevRes.GetSequence())
+		// suite.Require().Equal(account.GetAccountNumber(), prevRes.GetAccountNumber())
+		suite.Require().Equal(account, prevRes)
+
+		if checkPrevRes {
+			prevRes = account
+		}
+	}
+}
+
+func (suite *DeterministicTestSuite) TestGRPCQueryAccounts() {
+	rapid.Check(suite.T(), func(t *rapid.T) {
+		addr := testdata.AddrTestDeterministic(t)
+		suite.accountKeeper.SetAccount(suite.ctx,
+			suite.accountKeeper.NewAccountWithAddress(suite.ctx, addr))
```

Comment:
Let's modify this line to do the same as below, i.e. `types.NewAccount` and `authkeeper.SetAccount`